packages/middleware-signing/src/configurations.ts

```diff
@@ -45,20 +54,49 @@ export function resolveAwsAuthConfig<T>(
   let credentials =
     input.credentials || input.credentialDefaultProvider(input as any);
   const normalizedCreds = normalizeProvider(credentials);
-  const signingEscapePath = input.signingEscapePath || false;
-  const systemClockOffset = input.systemClockOffset || 0;
+  const {
+    signingEscapePath = false,
+    systemClockOffset = input.systemClockOffset || 0,
+    sha256
+  } = input;
+  let signer: Provider<RequestSigner>;
+  if (input.signer) signer = normalizeProvider(input.signer);
```

Comment:
This should be expanded out into a block form. Short form if statements are generally harder to read, and this one in particular is hard to read because the else uses a block but the first if statement doesn't.

I also think some comments in here to say what's going on would be useful to help future maintainers follow the code